sdk/core/azure-core-http-netty/src/main/java/com/azure/core/http/netty/NettyAsyncHttpClientBuilder.java

```diff
@@ -92,9 +93,9 @@ public NettyAsyncHttpClientBuilder(HttpClient nettyHttpClient) {
         if (this.baseHttpClient != null) {
             nettyHttpClient = baseHttpClient;
         } else if (this.connectionProvider != null) {
-            nettyHttpClient = HttpClient.create(this.connectionProvider);
+            nettyHttpClient = HttpClient.create(this.connectionProvider).resolver(DefaultAddressResolverGroup.INSTANCE);
         } else {
-            nettyHttpClient = HttpClient.create();
+            nettyHttpClient = HttpClient.create().resolver(DefaultAddressResolverGroup.INSTANCE);
```

Comment:
Do we know whether this default resolver is a blocking-resolver (that uses blocking InetSocketAddress resolution) or default is netty's non-blocking resolver?

Reply:
I remember browsing through this deadlock issue https://github.com/reactor/reactor-netty/issues/710 so thought of checking.

Reply:
So as per this - https://github.com/reactor/reactor-netty/issues/1563#issuecomment-871591339 there is no conclusion that blocking resolver can cause a dead-lock.

Btw if we explicitly set it for user, I guess user can still override it if needed right? (By providing an HttpClient instance)

Reply:
Yes, this would be overridable by using a configured `HttpClient`